powdr/src/lib.rs

```diff
@@ -103,6 +132,23 @@ impl Session {
     }
 }
 
+fn create_pil_file_path(asm_name: &Path) -> PathBuf {
+    let file_stem = asm_name.file_stem().unwrap().to_str().unwrap();
+    let opt_file_stem = format!("{file_stem}_opt");
+    asm_name.with_file_name(opt_file_stem).with_extension("pil")
+}
+
+fn read_file_if_exists(path: &Path) -> Option<String> {
+    if path.exists() {
```

Comment:
the `File::open` will fail exactly if the file doesn't exist, so i'd check its result instead...

Reply:
actually, there's a function in rust `std::fs::read_to_string` which could be used in place of `read_file_if_exists`

Reply:
replaced this function by `read_to_string`